scripts/west_commands/run_common.py

```diff
@@ -186,15 +186,13 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
     if not user_args.skip_rebuild:
         rebuild(command, build_dir, user_args)
 
-    if domains is None:
-        if user_args.domain is None:
-            # No domains are passed down and no domains specified by the user.
-            # So default domain will be used.
-            domains = [load_domains(build_dir).get_default_domain()]
-        else:
-            # No domains are passed down, but user has specified domains to use.
-            # Get the user specified domains.
-            domains = load_domains(build_dir).get_domains(user_args.domain)
+    if user_args.domain is None and command.name != "flash":
```

Comment:
`do_run_common()` should not know about specific commands.

Command specific handling should go into the dedicated command handler, in this case `flash.py`.

Reply:
Something more generic would be a `domain_args` parameter, where the flash command could pass:
`{"names": my_args.domain, "default_flash_order": True}`

and `do_run_common()` would have:
```py
if domain_args is None:
    ...
else:
    domains = load_domains(build_dir).get_domains(**domain_args)
```

Reply:
> Command specific handling should go into the dedicated command handler, in this case flash.py.

It can't go in flash.py hence why this PR exists because flash.py has an outdated content when flash.py runs

Reply:
> It can't go in flash.py

probably because this commit https://github.com/zephyrproject-rtos/zephyr/commit/a0267d2f4890427f10c16d510260e15a68bd1a84 completely broke the separation between the `do_run_common()` from the extension command specific handling (a commit part of a PR implementation I highly dis-approved).

So please cleanup and restore the `run_common.py` to be for the common stuff so that the extension specific `command.name != "flash"` can be correctly placed in `flash.py`.

Reply:
It has **nothing** to do with that commit, it is caused by [this](https://github.com/zephyrproject-rtos/zephyr/commit/eb95bed552f2977d77511d3c4fab0cba9f30a7a9 (and the interaction with sysbuild which was added years later), which was completely untouched and added 
```
    build_dir = get_build_dir(user_args)
    if not user_args.skip_rebuild:
        rebuild(command, build_dir, user_args)
```

Reply:
> It has nothing to do with that commit, it is caused by https://github.com/zephyrproject-rtos/zephyr/commit/eb95bed552f2977d77511d3c4fab0cba9f30a7a9

don't see the relevance. If `do_run()` is so flash specific, then it should handle all flash related python code directly in its own `do_run()` function and not use `do_run_common()`.

`do_run_common()` is intended for common stuff.
It is not required that `do_run()` calls `do_run_common()`, as an example the `do_run()` in `build.py` does not call `do_run_common()`:
https://github.com/zephyrproject-rtos/zephyr/blob/a35d5e78514fd7ae6f5f4b7b6cb5d3f443018508/scripts/west_commands/build.py#L169